src/plugins/mer/mervirtualboxmanager.cpp

```diff
@@ -634,12 +762,76 @@ VirtualMachineInfo virtualMachineInfoFromOutput(const QString &output)
             }
         } else if (rexp.cap(0).startsWith(QLatin1String("Session"))) {
             info.headless = rexp.cap(11) == QLatin1String("headless");
+        } else if (rexp.cap(0).startsWith(QLatin1String("\"SATA")) && fillVdiInfo) {
+            QString vdiPath = rexp.cap(12);
+            if (!vdiPath.isEmpty() && vdiPath != QLatin1String("none")) {
+                info.vdiPath = vdiPath;
+                fetchVdiInfo(&info);
+            }
+        } else if (rexp.cap(0).startsWith(QLatin1String("memory"))) {
+            int memorySize = rexp.cap(13).toInt();
+            if (memorySize > 0) {
+                info.memorySizeMb = memorySize;
+            }
+        } else if (rexp.cap(0).startsWith(QLatin1String("cpus"))) {
+            int cpu = rexp.cap(14).toInt();
+            if (cpu > 0) {
+                info.cpuCount = cpu;
+            }
         }
     }
 
     return info;
 }
 
+void fetchVdiInfo(VirtualMachineInfo *virtualMachineInfo)
+{
+    QStringList arguments;
+    arguments.append(QLatin1String(SHOWMEDIUMINFO));
+    arguments.append(virtualMachineInfo->vdiPath);
+    VBoxManageProcess process;
+    if (!process.runSynchronously(arguments))
+        return;
+
+    vdiInfoFromOutput(QString::fromLocal8Bit(process.readAllStandardOutput()), virtualMachineInfo);
+}
+
+void vdiInfoFromOutput(const QString &output, VirtualMachineInfo *virtualMachineInfo)
+{
+    // 1 vdi size
+    QRegExp rexp(QLatin1String("(?:Capacity:\\s+(\\d+)\\s+\\w+)"));
```

Comment:
Instead of "\\w+" we should clearly state we expect "MBytes" here, similarly to what you do in ramSizeFromOutput